lib/commands/connect/mapping-delete.js

```diff
@@ -12,19 +12,21 @@ module.exports = {
     {name: 'mapping'}
   ],
   flags: [
-    {name: 'resource', description: 'specific connection resource name', hasValue: true}
+    {name: 'resource', description: 'specific connection resource name', hasValue: true},
+    {name: 'confirm', hasValue: true}
   ],
   needsApp: true,
   needsAuth: true,
   run: cli.command(co.wrap(function* (context, heroku) {
-    let connection = yield api.withConnection(context, heroku);
-    let mapping = yield api.withMapping(connection, context.args.mapping);
-    let response = yield api.request(context.auth.password, 'DELETE', '/api/v3/mappings/' + mapping.id);
-    if (response.statusCode === 204) {
-      console.log('Mapping deleted.');
-    } else {
-      console.log('Received unexpected response during mapping delete ...');
-      api.printResponseError(response);
-    }
+    yield cli.confirmApp(context.app, context.flags.confirm);
+
+    yield cli.action('deleting mapping', co(function* () {
+      let connection = yield api.withConnection(context, heroku);
+      let mapping = yield api.withMapping(connection, context.args.mapping);
+      let response = yield api.request(context.auth.password, 'DELETE', '/api/v3/mappings/' + mapping.id);
+      if (response.statusCode !== 204) {
+        throw new Error(response.json.message || 'unknown error');
+      }
```

Comment:
As long as the action completes without error, it'll automatically print "done". So it'll look like this:

```
deleting mapping... done
```